### ISSUE TITLE
Sign in with Google temporarily disabled for this app [Fixed]

### ISSUE THREAD

**Issue by mdobydullah** [gdrive-org/gdrive#525]
Hi,
I'm using gdrive to backup my CentOS 7's files for a long time. Gdrive is the best among all google drive backup tools. A few days ago, I've launched a new server (CentOS 8) and tried to setup gdrive. But Google shows "Sign in with Google temporarily disabled for this app".

![error_sign_in](https://user-images.githubusercontent.com/13184472/72223461-1eba0f80-3599-11ea-9a70-7b2341b6c3a4.png)

Please take a look and please resume the maintenance. We love this tool.

**Update:** I've written a way to fix the issue. Please check [Mount Google Drive using GDrive on Linux Server with Own OAuth Credentials](https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials). I hope this method will work on all servers.

Thank you.

**Comment by mmmarceleza** [comment]
I had the same problem with a new server with Ubuntu 18.04.  :(

**Comment by hzb** [comment]
I had the same issue with Debian 9

**Comment by donand** [comment]
Same problem with Ubuntu 18.04, any update?

**Comment by mdobydullah** [comment]
Hi,

I've written a way to fix the issue:
https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials

**Comment by mdobydullah** [comment]
**Update:** I've written a way to fix the issue. Please check [Mount Google Drive using GDrive on Linux Server with Own OAuth Credentials](https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials). I hope this method will work on all servers.

**Comment by mdobydullah** [comment]
Hello Everyone,

I've written a way to fix the issue. Please check this article: [Mount Google Drive using GDrive on Linux Server with Own OAuth Credentials](https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials).

I hope this method will work on all servers.

Thanks.

**Comment by donand** [comment]
In the end I solved it by using rclone [](https://github.com/rclone/rclone), the login procedure is working there.

**Comment by mdobydullah** [comment]
Hi @donand 

Congrats. Rclone is a great alternative. Yesterday night, I've solved by compiling the GDrive project with my own OAuth credentials. Now it works fine. I'm using GDrive commands like before.

**Comment by ianxm** [comment]
I don't think this should be marked [Fixed]. Hard coding credentials is not secure. if someone copies your gdrive binary they'll have access to your account. Is this the recommended path? Will the root cause be fixed?

**Comment by mdobydullah** [comment]
Hi @ianxm,

GDrive is not being actively maintained at the moment. You should use the compiled gdrive only for you. For security reasons, you shouldn't share your compiled file with others.

**Comment by ianxm** [comment]
it was good of you to write up the steps to get it to work, but I don't think it's a good idea. It is a security risk even if you don't purposely share the file. It's uncommon to keep binaries around with credentials embedded in them. it would be easy to forget that file requires special handling. If you do this you might want to keep it in your `~/bin` and `chmod 700` it.

thanks for clarifying about the project. I didn't realize it was unmaintained. I'll use rclone.

**Comment by mdobydullah** [comment]
Hello @ianxm,

Thanks for your information.  Yeah [rclone](https://rclone.org/), [borg](https://github.com/borgbackup/borg), [odeke-em/drive](https://github.com/odeke-em/drive) are the great alternatives.

**Comment by develooper1994** [comment]
I had same issue and this fix doesn't work.
https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials

**Comment by mdobydullah** [comment]
Hi @develooper1994, Please follow all the steps properly and check file permission. It works for me and also tested by more users.

**Comment by cipri-tom** [comment]
@donand  have you managed to download a file with rclone which is not on your own drive ? `gdrive` had the `id` option for that. I can't figure out how to do it with rclone until `gdrive` is fixed

**Comment by develooper1994** [comment]
unfortunately but I don't need anymore.

**Comment by yarkm13** [comment]
No need to compile from source, just create service account in google console and use --service-account option

**Comment by mdobydullah** [comment]
Thanks @yarkm13 for sharing. I've tried some methods. Unfortunately, no method worked. Then I've compiled from source and works.

**Comment by yarkm13** [comment]
@mdobydullah you just need to go to https://console.cloud.google.com/iam-admin/serviceaccounts and create **service account** and you will be able to download json file with credentials (something like project-name-111-1b3f56a.json). Than put it to `~/.gdrive/`  and run _gdrive_

    ./gdrive --service-account project-name-111-1b3f56a.json about

**Comment by mdobydullah** [comment]
@yarkm13 thanks again. I'll try this method. :)

**Comment by yarkm13** [comment]
@mdobydullah i will be glad to hear from you if it was [in]success. Please share your experience

**Comment by mdobydullah** [comment]
@yarkm13 sure... :)

**Comment by captainheart** [comment]
> @mdobydullah you just need to go to https://console.cloud.google.com/iam-admin/serviceaccounts and create **service account** and you will be able to download json file with credentials (something like project-name-111-1b3f56a.json). Than put it to `~/.gdrive/` and run _gdrive_
> 
> ```
> ./gdrive --service-account project-name-111-1b3f56a.json about
> ```
> 
> such method was not mentioned in thread

I tried but uploaded file does not appear in my google drive and it's not able to view & download from browser.
The service account is the owner of the file.
What to do next?

**Comment by sanketssj5** [comment]
This works. Thank You

**Comment by mdobydullah** [comment]
@captainheart Sorry for late reply. Please [follow these steps](https://www.mynotepaper.com/mount-google-drive-using-gdrive-on-linux-server-with-own-oauth-credentials).

**Comment by mdobydullah** [comment]
@sanketssj5 Great. Nice to hear. :)

**Comment by tslater** [comment]
@nftleader I created a service account and did this...it uploaded...but now how do I know what account they uploaded to? the project is part of an organization. Do you know how to specify a user account?
